src/main/java/cs2103/duke/Storage.java

```diff
@@ -0,0 +1,85 @@
+package cs2103.duke;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+
+/**
+ * This class encapsulates a Storage object which saves the tasks that have been added by the user, it also
+ * performs the initialization for the duke.txt and its parent folder, if they are not already present.
+ */
+public class Storage {
+    private static ArrayList<Task> taskArrayList = new ArrayList<>();
+    private final String filePath;
+
+    public Storage(String dukeFilePath) {
+        this.filePath = dukeFilePath;
+    }
+
+    /**
+     * Checks the dukeFilePath for the existence of duke.txt, if the file or the folder containing it
+     * does not exist, create them.
+     *
+     * @throws IOException If an invalid input is detected.
+     */
+    public void initialize() throws IOException {
+        String dukeFileDirectoryPath = "./data";
+        String dukeFilePath = "./data/duke.txt";
+        File dukeFileDirectory = new File(dukeFileDirectoryPath);
+        File dukeFile = new File(dukeFilePath);
+        // creates directory if it does not exist
+        if (dukeFileDirectory.mkdir()) {
+            // dukeFile.mkdir();
+            System.out.println("folder: 'data/' has been created");
+        }
+        // creates file if it does not exist
+        if (dukeFile.createNewFile()) {
+            // dukeFile.createNewFile();
```

Comment:
Maybe this comment could be considered as redundant? The code line above it might be self-explanatory enough.